### ISSUE TITLE
Ignore some fields when converting JSON to CSV

### ISSUE THREAD

**Issue by Habush** [Cinchoo/ChoETL#42]
I am trying to write a JSON string into CSV file. I have defined the following POCO class  

```
public class Facility {
    [ChoJSONRecordField]
    public int? Id {get; set;}
    [ChoJSONRecordField]
    public string Name {get; set;}
    [ChoIgnoreMember] //Ignore Uuid
    public string Uuid {get; set;}
    [ChoJSONRecordField]
    public string CreatedAt {get; set;}
    [ChoJSONRecordField]
    public string UpdatedAt {get; set;}
    [ChoJSONRecordField]
    public bool Active {get; set;}
}
```

Here is the sample JSON string:

```
{"facilities":[{"id":39205, "name":"Sample1" ,"uuid":"ac2f3464-c425-4063-86ad-163521b1d610","createdAt":"2019-03-06T14:25:32Z","updatedAt":"2019-03-06T14:29:31Z","active":true},
{"id":35907,"name":"Sample2","uuid":"d371debb-f030-4c1e-b198-5eb562ceac0f","createdAt":"2019-02-21T09:33:25Z","updatedAt":"2019-02-21T09:33:25Z","active":true} ]}

```

I want to ignore `Uuid` field from being written in the CSV output. Here is how I am doing the conversion:

```
var memStream = new MemoryStream();
using (var r = ChoJSONReader<Facility>.LoadText(json).WithJSONPath("$..facilities")
.Configure(c => c.NullValue = "").Configure(c => c.IgnoredFields.Add("Uuid")))
{
    foreach (var facility in r)
    {
        var w = new ChoCSVWriter<Facility>(memStream).WithFirstLineHeader().Configure(c => 
               c.IgnoredFields.Add("Uuid"));
        w.Write(facility);
    }

    return memStream;
 }
```

However, the above code returns `memStream` without writting anything on it (`memStream.Length` equals `0`)

What's wrong with the above code? Or is there a better way to do this?

**Comment by Cinchoo** [comment]
Here is one way of doing it

```
        static void Issue42()
        {
            string json = @"{
    ""facilities"": [
        {
            ""id"": 39205,
            ""name"": ""Sample1"",
            ""uuid"": ""ac2f3464-c425-4063-86ad-163521b1d610"",
            ""createdAt"": ""2019-03-06T14:25:32Z"",
            ""updatedAt"": ""2019-03-06T14:29:31Z"",
            ""active"": true
        },
        {
            ""id"": 35907,
            ""name"": ""Sample2"",
            ""uuid"": ""d371debb-f030-4c1e-b198-5eb562ceac0f"",
            ""createdAt"": ""2019-02-21T09:33:25Z"",
            ""updatedAt"": ""2019-02-21T09:33:25Z"",
            ""active"": true
        }
    ]
}
";
            StringBuilder csv = new StringBuilder();
            using (var p = ChoJSONReader<Facility>.LoadText(json)
                .WithJSONPath("$..facilities")
                )
            {
                using (var w = new ChoCSVWriter<Facility>(csv)
                    .WithFirstLineHeader()
                    )
                {
                    w.Write(p);
                }
            }

            Console.WriteLine(csv.ToString());
        }
```


Output:

```
Id,Name,CreatedAt,UpdatedAt,Active
39205,Sample1,03/06/2019 14:25:32,03/06/2019 14:29:31,1
35907,Sample2,02/21/2019 09:33:25,02/21/2019 09:33:25,1
```

**Comment by Habush** [comment]
Your suggestion works! But is there any reason for it not working when passing the `Stream` instead of a `StringBuilder`?

**Comment by Cinchoo** [comment]
Got it, found out that the framework is closing stream on dispose. Put a fix on version 1.1.0.0

Here is working sample:

```
            string json = @"{
    ""facilities"": [
        {
            ""id"": 39205,
            ""name"": ""Sample1"",
            ""uuid"": ""ac2f3464-c425-4063-86ad-163521b1d610"",
            ""createdAt"": ""2019-03-06T14:25:32Z"",
            ""updatedAt"": ""2019-03-06T14:29:31Z"",
            ""active"": true
        },
        {
            ""id"": 35907,
            ""name"": ""Sample2"",
            ""uuid"": ""d371debb-f030-4c1e-b198-5eb562ceac0f"",
            ""createdAt"": ""2019-02-21T09:33:25Z"",
            ""updatedAt"": ""2019-02-21T09:33:25Z"",
            ""active"": true
        }
    ]
}
";
            var memStream = new MemoryStream();
            using (var p = ChoJSONReader<Facility>.LoadText(json)
                .WithJSONPath("$..facilities")
                )
            {
                using (var w = new ChoCSVWriter<Facility>(memStream)
                    .WithFirstLineHeader()
                    )
                {
                    w.Write(p);
                }
            }

            Console.WriteLine(Encoding.ASCII.GetString(memStream.ToArray()));

```

Output:

```
Id,Name,CreatedAt,UpdatedAt,Active
39205,Sample1,03/06/2019 14:25:32,03/06/2019 14:29:31,1
35907,Sample2,02/21/2019 09:33:25,02/21/2019 09:33:25,1
```
